torch/_dynamo/variables/functions.py

```diff
@@ -82,6 +82,16 @@ def call_function(
             self, list(self.self_args()) + list(args), kwargs
         )
 
+    def call_hasattr(self, tx, name: str) -> VariableTracker:
+        result = False
+
+        try:
+            result = hasattr(self.get_function(), name)
+        except NotImplementedError:
+            if name == "__name__" and isinstance(self, NestedUserFunctionVariable):
```

Comment:
Even better, why not add `__name__` to the object returned by `get_function`?

Reply:
Calling `get_function` on a `NestedUserFunctionVariable` may raise `NotImplementedError` if the nested function has a closure. See:

https://github.com/pytorch/pytorch/blob/95a86ed9ca107329151e0dc172386d50dd3471c6/torch/_dynamo/variables/functions.py#L402-L405